pandas/io/formats/style.py

```diff
@@ -204,6 +206,91 @@ def _repr_html_(self) -> str:
         """
         return self.render()
 
+    def set_tooltips(self, ttips: DataFrame) -> "Styler":
+        """
+        Add string based tooltips that will appear in the `Styler` HTML result. These
+        tooltips are applicable only to`<td>` elements.
+
+        Parameters
+        ----------
+        ttips : DataFrame
+            DataFrame containing strings that will be translated to tooltips, mapped
+            by identical column and index values that must exist on the underlying
+            `Styler` data. None, NaN values, and empty strings will be ignored and
+            not affect the rendered HTML.
+
+        Returns
+        -------
+        self : Styler
+
+        Notes
+        -----
+        Tooltips are created by adding `<span class="pd-t"></span>` to each data cell
+        and then manipulating the table level CSS to attach pseudo hover and pseudo
+        after selectors to produce the required the results. For styling control
+        see `:meth:Styler.set_tooltips_class`.
+        Tooltips are not designed to be efficient, and can add large amounts of
+        additional HTML for larger tables, since they also require that `cell_ids`
+        is forced to `True`.
+
+        Examples
+        --------
+        >>> df = pd.DataFrame(data=[[0, 1], [2, 3]])
+        >>> ttips = pd.DataFrame(
+        ...    data=[["Min", ""], [np.nan, "Max"]], columns=df.columns, index=df.index
+        ... )
+        >>> s = Styler(df, uuid="_").set_tooltips(ttips).render()
+        """
+        if not self.cell_ids:
+            # tooltips not optimised for individual cell check.
+            raise NotImplementedError(
+                "Tooltips can only render with 'cell_ids' is True."
+            )
+        self.tooltips.tt_data = ttips
+        return self
+
+    def set_tooltips_class(
+        self,
+        name: Optional[str] = None,
+        properties: Optional[List[Tuple[str, Union[str, int, float]]]] = None,
```

Comment:
```suggestion
        properties: Optional[Sequence[Tuple[str, Union[str, int, float]]]] = None,
```

Or does this have to be a List?

Reply:
The underlying `Styler` `table_styles` get extended with these `_Tooltips` `table_styles` so I have maintained the same type definition for consistency.

Although the `extend()` method would work on sequence as well. Happy to take your lead on this one... Shall I change to sequence?

Reply:
Yea let's change to Sequence - typically our API is typed as permissive as possible when it comes to container objects